Plugins/NativeEngine/Source/ShaderCache.cpp

```diff
@@ -0,0 +1,180 @@
+#include <Babylon/ShaderCache.h>
+#include <map>
+#include "ShaderCompiler.h"
+#include "ShaderCache.h"
+
+namespace
+{
+    void SaveString(std::ofstream& stream, const std::string& string)
+    {
+        uint32_t stringSize{static_cast<uint32_t>(string.size())};
+        stream.write(reinterpret_cast<const char*>(&stringSize), sizeof(uint32_t));
+        stream.write(reinterpret_cast<const char*>(string.data()), string.size());
+    }
+
+    void LoadString(std::ifstream& stream, std::string& string)
+    {
+        uint32_t stringSize;
+        stream.read(reinterpret_cast<char*>(&stringSize), sizeof(uint32_t));
+        string.resize(stringSize);
+        stream.read(string.data(), stringSize);
+    }
+
+    struct ShaderHash
+    {
+        size_t vertexHash;
+        size_t fragmentHash;
+        bool operator < (const ShaderHash& other) const {
+            if (vertexHash < other.vertexHash || (vertexHash == other.vertexHash && fragmentHash < other.fragmentHash))
+                return true;
+            return false;
+        }
+    };
```

Comment:
Seems this is in both the .h and the .cpp

Reply:
fixed